### ISSUE TITLE
NOMADS HRRR and HIRES-ARW-ALASKA are failing

### ISSUE THREAD

**Issue by nwagenbrenner** [firelab/windninja#248]
This is happening at least intermittently.

Time: 2017-03-30 02:00:01, cli_NOMADS-HRRR-CONUS-3-KM.cfg: Failed, stderr = ERROR 1: HTTP error code : 404
ERROR 1: Failed to download file.
Warning 1: Failed to download forecast, stepping back one forecast run time step.
ERROR 1: Operation timed out after 20001 milliseconds with 0 bytes received
ERROR 1: Failed to download file.
ERROR 1: Operation timed out after 20001 milliseconds with 0 bytes received
ERROR 1: Failed to download file.
Warning 1: Threaded download failed, attempting serial download for hrrr.t07z.wrfsfcf02.grib2
Warning 1: Threaded download failed, attempting serial download for hrrr.t07z.wrfsfcf04.grib2
ERROR 1: HTTP error code : 404
ERROR 1: Failed to download file.
ERROR 1: HTTP error code : 404
ERROR 1: Failed to download file.
ERROR 1: HTTP error code : 404
ERROR 1: Failed to download file.
ERROR 1: HTTP error code : 404
ERROR 1: Failed to download file.
Warning 1: Threaded download failed, attempting serial download for hrrr.t07z.wrfsfcf05.grib2
ERROR 1: HTTP error code : 404
ERROR 1: Failed to download file.
Warning 1: Threaded download failed, attempting serial download for hrrr.t07z.wrfsfcf06.grib2
ERROR 1: HTTP error code : 404
ERROR 1: Failed to download file.
Warning 1: Threaded download failed, attempting serial download for hrrr.t07z.wrfsfcf07.grib2
ERROR 1: HTTP error code : 404
ERROR 1: Failed to download file.
Warning 1: Threaded download failed, attempting serial download for hrrr.t07z.wrfsfcf08.grib2
ERROR 1: HTTP error code : 404
ERROR 1: Failed to download file.
Warning 1: Failed to download forecast, stepping back one forecast run time step.
ERROR 1: Operation timed out after 20001 milliseconds with 0 bytes received
ERROR 1: Failed to download file.
ERROR 1: Operation timed out after 20001 milliseconds with 19850 bytes received
ERROR 1: Failed to download file.
Warning 1: Threaded download failed, attempting serial download for hrrr.t06z.wrfsfcf10.grib2
Warning 1: Threaded download failed, attempting serial download for hrrr.t06z.wrfsfcf12.grib2
ERROR 1: Operation timed out after 20000 milliseconds with 0 bytes received
ERROR 1: Failed to download file.
Warning 1: Failed to download forecast, stepping back one forecast run time step.

Time: 2017-03-30 02:00:01, cli_NOMADS-HIRES-ARW-ALASKA-5-KM.cfg: Failed, stderr = ERROR 1: HTTP error code : 404
ERROR 1: Failed to download file.
Warning 1: Failed to download forecast, stepping back one forecast run time step.
ERROR 1: Operation timed out after 20000 milliseconds with 8263 bytes received
ERROR 1: Failed to download file.
Warning 1: Threaded download failed, attempting serial download for hiresw.t06z.arw_5km.f06.ak.grib2
ERROR 1: Operation timed out after 20000 milliseconds with 8263 bytes received
ERROR 1: Failed to download file.
Warning 1: Threaded download failed, attempting serial download for hiresw.t06z.arw_5km.f10.ak.grib2
ERROR 1: Operation timed out after 20000 milliseconds with 8263 bytes received
ERROR 1: Failed to download file.
Warning 1: Failed to download forecast, stepping back one forecast run time step.

Time: 2017-03-30 02:00:01, cli_NOMADS-NAM-CONUS-12-KM.cfg: Failed, stderr = ERROR 1: HTTP error code : 500
ERROR 1: HTTP error code : 500
ERROR 1: HTTP error code : 500
ERROR 1: HTTP error code : 500
ERROR 1: HTTP error code : 500
ERROR 1: HTTP error code : 500
ERROR 1: HTTP error code : 500
ERROR 1: HTTP error code : 500
ERROR 1: HTTP error code : 500
ERROR 1: HTTP error code : 500
ERROR 1: HTTP error code : 500
ERROR 1: HTTP error code : 500
ERROR 1: HTTP error code : 500
ERROR 1: HTTP error code : 500
ERROR 1: HTTP error code : 500
ERROR 1: HTTP error code : 500
ERROR 1: HTTP error code : 500
ERROR 1: HTTP error code : 500
ERROR 4: `/vsizip/NINJA_FCST_16312_2.zip/nam.t06z.awphys00.grb2.tm00' not recognized as a supported file format.

Time: 2017-03-30 02:00:01, cli_NOMADS-NAM-ALASKA-11.25-KM.cfg: Failed, stderr = ERROR 1: HTTP error code : 500
ERROR 1: HTTP error code : 500
ERROR 1: HTTP error code : 500
ERROR 1: HTTP error code : 500
ERROR 1: HTTP error code : 500
ERROR 1: HTTP error code : 500
ERROR 4: `/vsizip/NINJA_FCST_16455_2.zip/nam.t06z.awak3d00.grb2.tm00' not recognized as a supported file format.

**Comment by ksshannon** [comment]
I think the timeouts are associated with recent performance issues on the NOMADS side (from the mailing list):

> NOMADS and NCEP FTP users,
> 
> We are again seeing slow performance on various NCEP websites, including
> NOMADS and the NCEP FTP model dissemination site. Our system administrators
> identified a resource constraint and have implemented a fix. You should now
> begin to see improvements in the performance of the NOMADS and NCEP FTP
> sites.
> 
> We apologize for impacts these slowdowns have caused.

The 500 errors should be fixed in the tin-16-41 branch.  I'll merge it and push and we'll see how it goes.

**Comment by ksshannon** [comment]
Merged.  If the CI passes, the only other issue is on the NOMADS FTP side.  Even the web interface is very slow.  I guess the only thing we can do there is have a better error message?

**Comment by ksshannon** [comment]
@nwagenbrenner when do your nightly tests run?

**Comment by ksshannon** [comment]
More NOMADS issues:

> Fabio,
> 
> We are experiencing slowness with multiple NCEP websites and it was decided at 18:45Z to fail our NOMADS service to our secondary site. You should see improved performance since the fail-over.
> 
> We apologize for the inconvenience.
> 
> Justin Cooke
> NCEP Central Operations
> 
> On Tue, Mar 28, 2017 at 2:31 PM, Fabio F.Gervasi <fabio.gervasi70@gmail.com> wrote:
> 
>     Hi,
> 
>     Since today 12Z 28/03/2017 I can not download data or connect to the site http://nomads.ncep.noaa.gov (the browser responds to me "impossible to reach the site"). It happens from any IP and/or network that I use to connect.
> 
>     Any news?
> 
>     Thanks,
>     Fabio.
> 
>     _______________________________________________
>     Ncep.list.pmb-dataflow mailing list
>     Ncep.list.pmb-dataflow@lstsrv.ncep.noaa.gov
>     https://www.lstsrv.ncep.noaa.gov/mailman/listinfo/ncep.list.pmb-dataflow
> 
> 
> 
> _______________________________________________
> NCEP.List.Nomads-Users mailing list
> NCEP.List.Nomads-Users@lstsrv.ncep.noaa.gov
> https://www.lstsrv.ncep.noaa.gov/mailman/listinfo/ncep.list.nomads-users

But not the whole problem.  I'll look into it when I get some time.

Associated with #244.

**Comment by nwagenbrenner** [comment]
The nightly tests run at 8:00 UTC. A better error message might be good if this is likely to be a recurring issue. At least that way users will know it's an issue on the wx model server.

**Comment by ksshannon** [comment]
Okay, I'll look at returning finer-grained errors.

**Comment by ksshannon** [comment]
I don't think we can make a determination on this until NOMADS is back up reliably.  I think they are all just flaky server issues.  I was able to just download a HRRR, so the data is okay.  The web interface is still super slow.
